### PR TITLE
implement root call and adding help information

### PR DESCRIPTION
**Pull request by majia67**
This PR implements the root call of the inventory API. Basically a help info of the API will be returned. The user may expect a response like the following screenshot:

image

### CODE REVIEW

File: server.py [hunk 1]
```diff
@@ -63,8 +63,21 @@ def internal_server_error(error):
...
-    return jsonify(name='Inventory Root'), HTTP_200_OK
+    """ Return help information about the API """
+    return jsonify(
+        name='Inventory Management REST API Service',
+        version='1.0',
+        usage={
+            'help': 'GET /',
+            'list all': 'GET /inventory',
+            'read': 'GET /inventory/{prod_id}',
+            'add': 'POST /inventory',
```
Comment (jockyxu1122) [inline server.py]: Since we are calling this `create` everywhere else. Change this to `create`?

Comment (majia67) [reply server.py]: Sure!

**Comment by MingshiPeng** [comment]
Just a minor issue here, is there a way to not to return the "usage" part alphabetically (which is a little bit of unfriendly) so it might look nicer?

**Comment by majia67** [comment]
@MingshiPeng items in a dictionary does not have a specific order. List preserves the order, but it does not read as friendly as the dictionary.

**Pull request merged**